doc/examples/features_detection/plot_haar.py

```diff
@@ -0,0 +1,101 @@
+Haar-like feature descriptor
+============================
+
+Haar-like feature descriptors are a set of simple features which was used in
+the first real-time face detector [1]_. Indeed, these features can be
+efficiently computed using a so-called integral image and by subtracting
+rectangle area. The following example will show the mechanism to build this
+family of descriptors.
+
+References
+----------
+
+.. [1] Viola, Paul, and Michael J. Jones. "Robust real-time face
+       detection." International journal of computer vision 57.2
+       (2004): 137-154.
+       http://www.merl.com/publications/docs/TR2004-043.pdf
+       DOI: 10.1109/CVPR.2001.990517
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from skimage.data import astronaut
+
+from skimage.color import rgb2gray
+
+from skimage.io import imshow
+from skimage.io import imshow_collection
+
+from skimage.feature import haar_like_feature
+from skimage.feature import haar_like_feature_coord
+from skimage.feature import draw_haar_like_feature
+
+###############################################################################
+# The different Haar-like feature descriptors
+###############################################################################
+# The Haar-like feature descriptors come into 5 different types as illustrated
+# in the figure below. The value of the descriptor consists in the subtraction
+# between the green and red rectangles.
+
+images = [np.zeros((2, 2)), np.zeros((2, 2)),
+          np.zeros((3, 3)), np.zeros((3, 3)),
+          np.zeros((2, 2))]
+
+feature_types = ['type-2-x', 'type-2-y',
+                 'type-3-x', 'type-3-y',
+                 'type-4']
+
+feature = [draw_haar_like_feature(img, 0, 0, img.shape[1], img.shape[0],
+                                  feat_t, max_n_features=1)
+           for img, feat_t in zip(images, feature_types)]
+
+fig, axs = plt.subplots(3, 2)
+for ax, img, feat_t in zip(np.ravel(axs), images, feature_types):
+    ax.imshow(draw_haar_like_feature(img, 0, 0, img.shape[1], img.shape[0],
+                                     feat_t, max_n_features=1,
+                                     random_state=0))
+    ax.set_title(feat_t)
+fig.suptitle('The different Haar-like feature descriptors')
+plt.axis('off')
+plt.tight_layout()
+
+###############################################################################
+# The usual feature extraction scheme
+###############################################################################
+# The procedure to extract the Haar-like feature for an image is quite easy:
+# a region of interest (ROI) is defined for which all possible feature will
+# be extracted. In order to compute those features, an integral image needs to
+# be provided. Typically, for a ROI of 24 by 24 pixels, the total amount of
+# features which can be extracted is 160,000.
+
+image = rgb2gray(astronaut())
+plt.figure()
+plt.imshow(draw_haar_like_feature(image, 125, 200, 50, 50,
```

Comment:
I couldn't figure out, looking at this ROI, *why* the features got selected there.  Any insight?

Reply:
Those features have not been selected. We don't make any selection, we are only generating them. I just plotted 3 random features over the 20,000 possible computed in the ROI of type-4.

Reply:
Seeing your question, it means that my example is not good (I was already questioning myself :))
What do you have in mind knowing that basically, all features are extracted.

Reply:
FYI: I really like this: https://www.youtube.com/watch?v=hPCTwxF0qf4
For this case only the selected features are shown.

However, I think that it would be interesting to show something similar. take the example of the first extracted feature, and show the computation, then a second feature, ... etc. until the point that it is clear that all possible features are extracted for the ROI.